perfkitbenchmarker/virtual_machine.py

```diff
@@ -285,36 +314,65 @@ def RemoteCopy(self, file_path, remote_path='', copy_to=True):
                     (retcode, full_cmd, stdout, stderr))
       raise errors.VmUtil.SshConnectionError(error_text)
 
+  def _PushLongRunningCommandScripts(self):
```

Comment:
`PushDataFile` and `PushFile` are generic functions for pushing files to the remote VM.
This function defers most of the work `PushDataFile`, under a lock so this work is done only once.  I don't think it adds a lot of generic value.